tests/test_arithmetics.rs

```diff
@@ -292,46 +328,182 @@ impl PyNumberProtocol for LhsAndRhsArithmetic {
         format!("{:?} - RA", other)
     }
 
-    fn __rpow__(&self, other: &PyAny, _mod: Option<&'p PyAny>) -> String {
-        format!("{:?} ** RA", other)
+    fn __rmul__(&self, other: &PyAny) -> String {
+        format!("{:?} * RA", other)
     }
 
-    fn __add__(lhs: &PyAny, rhs: &PyAny) -> String {
-        format!("{:?} + {:?}", lhs, rhs)
+    fn __rlshift__(&self, other: &PyAny) -> String {
+        format!("{:?} << RA", other)
     }
 
-    fn __sub__(lhs: &PyAny, rhs: &PyAny) -> String {
-        format!("{:?} - {:?}", lhs, rhs)
+    fn __rrshift__(&self, other: &PyAny) -> String {
+        format!("{:?} >> RA", other)
     }
 
-    fn __pow__(lhs: &PyAny, rhs: &PyAny, _mod: Option<u32>) -> String {
-        format!("{:?} ** {:?}", lhs, rhs)
+    fn __rand__(&self, other: &PyAny) -> String {
+        format!("{:?} & RA", other)
+    }
+
+    fn __rxor__(&self, other: &PyAny) -> String {
+        format!("{:?} ^ RA", other)
+    }
+
+    fn __ror__(&self, other: &PyAny) -> String {
+        format!("{:?} | RA", other)
+    }
+
+    fn __rpow__(&self, other: &PyAny, _mod: Option<&'p PyAny>) -> String {
+        format!("{:?} ** RA", other)
     }
 }
 
 #[pyproto]
-impl PyObjectProtocol for LhsAndRhsArithmetic {
+impl PyObjectProtocol for LhsOverridesRhs {
     fn __repr__(&self) -> &'static str {
         "BA"
     }
 }
 
 #[test]
-fn lhs_override_rhs() {
+fn lhs_overrides_rhs() {
     let gil = Python::acquire_gil();
     let py = gil.python();
 
-    let c = PyCell::new(py, LhsAndRhsArithmetic {}).unwrap();
+    let c = PyCell::new(py, LhsOverridesRhs {}).unwrap();
     // Not overrided
     py_run!(py, c, "assert c.__radd__(1) == '1 + RA'");
     py_run!(py, c, "assert c.__rsub__(1) == '1 - RA'");
+    py_run!(py, c, "assert c.__rmul__(1) == '1 * RA'");
+    py_run!(py, c, "assert c.__rlshift__(1) == '1 << RA'");
+    py_run!(py, c, "assert c.__rrshift__(1) == '1 >> RA'");
+    py_run!(py, c, "assert c.__rand__(1) == '1 & RA'");
+    py_run!(py, c, "assert c.__rxor__(1) == '1 ^ RA'");
+    py_run!(py, c, "assert c.__ror__(1) == '1 | RA'");
     py_run!(py, c, "assert c.__rpow__(1) == '1 ** RA'");
     // Overrided
     py_run!(py, c, "assert 1 + c == '1 + BA'");
     py_run!(py, c, "assert 1 - c == '1 - BA'");
+    py_run!(py, c, "assert 1 * c == '1 * BA'");
+    py_run!(py, c, "assert 1 << c == '1 << BA'");
+    py_run!(py, c, "assert 1 >> c == '1 >> BA'");
+    py_run!(py, c, "assert 1 & c == '1 & BA'");
+    py_run!(py, c, "assert 1 ^ c == '1 ^ BA'");
+    py_run!(py, c, "assert 1 | c == '1 | BA'");
     py_run!(py, c, "assert 1 ** c == '1 ** BA'");
 }
 
+#[pyclass]
+#[derive(Debug)]
+struct LhsFellbackToRhs {}
```

Comment:
Rather than making a new `#[pyclass]` here, was it possible to just use the renamed `LhsAndRhsArithmetic` class? There's four sets of tests which we can write:

```
#[test]
fn can_call_lhs_explicitly() {
    py_run!(py, c, "assert c.__add__(1) == '1 + BA'");
}

#[test]
fn can_call_rhs_explicitly() {
    py_run!(py, c, "assert c.__radd__(1) == '1 + RA'");
    // etc for all operators
}

#[test]
fn can_call_lhs_operation() {
     py_run!(py, c, "assert c + 1 == '1 + BA'");
    // etc for all operators
}

#[test]
fn can_call_rhs_operation() {
     py_run!(py, c, "assert 1 + c == '1 + RA'");
    // etc for all operators
}
```

Reply:
Thanks, I simplified the test cases, though I still stuck with `lhs_overrides_rhs` and `lhs_fellback_to_rhs`.